### PR TITLE
Add intial wasm support to the UEFI and bare-metal runtimes

### PR DESCRIPTION
**Pull request by conradgrobler**
We can't currently send the Wasm module bytes to the runtime from the loader, so the compiled `echo` Wasm module is embedded into the runtime and loaded from the embedded bytes for now.

### CODE REVIEW

File: oak_functions/wasm/src/lib.rs [hunk 1]
```diff
@@ -463,7 +463,12 @@ where
     }
 
     pub fn handle_invoke(&self, request: Request) -> anyhow::Result<Response> {
-        let request_bytes = request.body;
+        let response_bytes = self.handle_raw_invoke(request.body)?;
+        Ok(Response::create(StatusCode::Success, response_bytes))
+    }
```
Comment (jul-sh) [inline oak_functions/wasm/src/lib.rs:468]: Do you have any thoughts whether we plan to eventually align the the two runtimes response handling? I think there's good arguments against using the current runtime's proto wrapper in the UEFI runtime, but also eventual consistency would be nice.

File: experimental/uefi/runtime/src/framing.rs [hunk 2]
```diff
@@ -14,26 +14,13 @@
...
-    // An error occured while deserializing.
-    Deserialization(ciborium::de::Error<T>),
-
-    // An error occured while serializing.
-    Serialization(ciborium::ser::Error<T>),
-
-    // An error occured in remote attestation.
-    Attestation(anyhow::Error),
-}
-
```
Comment (jul-sh) [inline experimental/uefi/runtime/src/framing.rs:34]: One thing I liked about this struct is facilitating differentiation between types of errors that may either be private or could be forwarded to the client. We can properly tackle that as part of #2783 though, for now using anyhow consistently seems nice. :)

**Review by jul-sh — APPROVED** [review]
thank you. lgtm!

File: experimental/uefi/runtime/src/framing.rs (hunk 2 above)
Comment (conradgrobler) [reply experimental/uefi/runtime/src/framing.rs:34]: I agree with the idea about deciding which error types should be reported to the client, but I don't think this enum provided a meaningful distinction. I agree that this should be addressed in a more systematic way as part of #2783 and #2794.

File: oak_functions/wasm/src/lib.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_functions/wasm/src/lib.rs:468]: I think the plan is to eventually align the two runtimes once we have settled on an approach to framing and an IDL.

**Review by conradgrobler — COMMENTED** [review]
Thanks for the review.

File: experimental/uefi/app/src/main.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@
 #![test_runner(crate::test_runner)]
 #![reexport_test_harness_main = "test_main"]
 
-use runtime::echo;
+use runtime::framing;
```
Comment (andrisaar) [inline experimental/uefi/app/src/main.rs:27]: I will have to admit that I'd much rather see the (a) renaming of `echo` to `framing` and (b) refactoring the error handling in a separate PR, but it's not like Github makes that easy.

**Review by andrisaar — APPROVED** [review]

**Pull request merged**